src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -934,6 +1116,205 @@ - (void)readAttributesWithEndpointID:(NSNumber * _Nullable)endpointID
+                   };
+
+                   MTRReadClientContainer * container = [[MTRReadClientContainer alloc] init];
+                   container.deviceID = self.nodeID;
+
+                   std::vector<app::AttributePathParams> attributePathParamsList;
+                   std::vector<app::EventPathParams> eventPathParamsList;
+
+                   if (attributePaths != nil) {
+                       for (uint8_t i = 0; i < [attributePaths count]; i++) {
+                           chip::EndpointId endpointId = kInvalidEndpointId;
+                           chip::ClusterId clusterId = kInvalidClusterId;
+                           chip::AttributeId attributeId = kInvalidAttributeId;
+
+                           if ([attributePaths[i] endpoint]) {
+                               endpointId = static_cast<chip::EndpointId>([[attributePaths[i] endpoint] unsignedShortValue]);
+                           }
+
+                           if ([attributePaths[i] cluster]) {
+                               clusterId = static_cast<chip::ClusterId>([[attributePaths[i] cluster] unsignedLongValue]);
+                           }
+
+                           if ([attributePaths[i] attribute]) {
+                               attributeId = static_cast<chip::AttributeId>([[attributePaths[i] attribute] unsignedLongValue]);
+                           }
+
+                           attributePathParamsList.push_back(app::AttributePathParams(endpointId, clusterId, attributeId));
+                       }
+                       container.pathParamsSize = attributePathParamsList.size();
+                       container.pathParams = attributePathParamsList.data();
+                   }
+                   if (eventPaths != nil) {
+                       for (uint8_t i = 0; i < [eventPaths count]; i++) {
+                           chip::EndpointId endpointId = kInvalidEndpointId;
+                           chip::ClusterId clusterId = kInvalidClusterId;
+                           chip::EventId eventId = kInvalidEventId;
+
+                           if ([eventPaths[i] endpoint]) {
+                               endpointId = static_cast<chip::EndpointId>([[eventPaths[i] endpoint] unsignedShortValue]);
+                           }
+
+                           if ([eventPaths[i] cluster]) {
+                               clusterId = static_cast<chip::ClusterId>([[eventPaths[i] cluster] unsignedLongValue]);
+                           }
+
+                           if ([eventPaths[i] event]) {
+                               eventId = static_cast<chip::EventId>([[eventPaths[i] event] unsignedLongValue]);
+                           }
+
+                           eventPathParamsList.push_back(app::EventPathParams(endpointId, clusterId, eventId));
+                       }
+                       container.eventPathParamsSize = eventPathParamsList.size();
+                       container.eventPathParams = eventPathParamsList.data();
```

Comment:
Again, this is going to break.  How was this tested?  There should probably be some unit tests for this as part of this PR....

Reply:
It's hard to agree with your opinion.
Even if it is passed to the stack, it is changed from SendRequest to TLV Format, so there is no problem even if it is made into a stack.
 
(Of course, free should be deleted below.)

Reply:
I'm sorry. I checked this variable has to make `new` and `delete`.